src/arrays/arraymath.jl

```diff
@@ -61,6 +61,9 @@ end
 
 *(dm1::DualMatrix, dm2::DualMatrix) = (dm1.qarr*dm2.qarr)'
 
++(qarr1::Union(QuArray,CTranspose), qarr2::Union(QuArray,CTranspose)) = bases(qarr1) == bases(qarr2) ? (return QuArray(coeffs(qarr1)+coeffs(qarr2), bases(qarr1))) : "Not compatible"
```

Comment:
I think you want to use `error("Bases not compatible.")` here. Should we use type parameters (for the basis and `N`) to restrict the functions in the first place?